modules/gitops-configuring-gitops-managed-resources-to-use-mounted-secrets.adoc

```diff
@@ -0,0 +1,106 @@
+// Module is included in the following assemblies:
+//
+// * securing_openshift_gitops/managing-secrets-securely-using-sscsid-with-gitops.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="gitops-configuring-gitops-managed-resources-to-use-mounted-secrets_{context}"]
+= Configuring {gitops-shortname} managed resources to use mounted secrets
+
+You must configure the {gitops-shortname} managed resources by adding volume mounts configuration to a deployment and configure the container pod to use the mounted secret. For example, consider that you want to add volume mounts configuration to the deployment of `app-taxi` application and the `100-deployment.yaml` file is in the `/environments/dev/apps/app-taxi/services/taxi/base/config/` directory.
+
+.Prerequisites
+
+* You have the AWS Secrets Manager resources stored in your {gitops-shortname} repository.
+* You have the Secrets Store Container Storage Interface (SSCSI) driver configured to mount secrets from AWS Secrets Manager.
+
+.Procedure
+
+. Add volume mounts configuration in the target resource:
+
+.. Add the volume mounting to the deployment YAML file and configure the container pod to use the mounted secret:
++
+.Example deployment file to use the secret provider class resources
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: taxi                                
+  namespace: dev                            # <1>
+spec:
+  replicas: 1
+  template:
+    metadata:
+# ...
+    spec:
+      containers:
+        - image: nginxinc/nginx-unprivileged:latest
+          imagePullPolicy: Always
+          name: taxi
+          ports:
+            - containerPort: 8080
+          volumeMounts:
+            - name: secrets-store-inline
+              mountPath: "/mnt/secrets-store"   # <2>
+              readOnly: true
+          resources: {}
+    serviceAccountName: default
+    volumes:
+      - name: secrets-store-inline
+        csi:
+          driver: secrets-store.csi.k8s.io
+          readOnly: true
+          volumeAttributes:
+            secretProviderClass: "my-aws-provider" # <3>
+    status: {}
+# ...
+----
+<1> Namespace for the deployment. This must be the same namespace as the secret provider class.
+<2> The path to mount secrets in the volume mount.
+<3> Name of the secret provider class.
+
+.. Push the updated resource YAML file to your {gitops-shortname} repository.
+
+. In the Argo CD UI, click *REFRESH* on the target application page to apply the updated deployment manifest.
+
+. Verify that all the resources are successfully synchronized on the target application page.
+
+. Verify that you can you can access the secrets from AWS Secrets manager in the pod volume mount:
+
+.. List the secrets in the pod mount: 
++
+[source,terminal]
+----
+$ oc exec <deployment-name>-<hash> -n <my-namespace> -- ls /mnt/secrets-store/
```

Comment:
Just curious, is `<deployment-name>-<hash>` a common term used in documentation?

Reply:
@varshab1210 this is a variable to be replaced by the user. 

Refer to the the RH OCP Doc guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference